src/wallet/bdb.cpp

```diff
@@ -626,44 +626,36 @@ void BerkeleyEnvironment::Flush(bool fShutdown)
 
 bool BerkeleyBatch::PeriodicFlush(BerkeleyDatabase& database)
 {
-    if (database.IsDummy()) {
-        return true;
-    }
-    bool ret = false;
+    // There's nothing to do for dummy databases. Return true.
+    if (database.IsDummy()) return true;
+
     BerkeleyEnvironment *env = database.env.get();
     const std::string& strFile = database.strFile;
+
+    // Don't flush if we can't acquire the lock.
     TRY_LOCK(cs_db, lockDb);
-    if (lockDb)
-    {
-        // Don't do this if any databases are in use
-        int nRefCount = 0;
-        std::map<std::string, int>::iterator mit = env->mapFileUseCount.begin();
-        while (mit != env->mapFileUseCount.end())
-        {
-            nRefCount += (*mit).second;
-            mit++;
-        }
+    if (!lockDb) return false;
 
-        if (nRefCount == 0)
-        {
-            std::map<std::string, int>::iterator mi = env->mapFileUseCount.find(strFile);
-            if (mi != env->mapFileUseCount.end())
-            {
-                LogPrint(BCLog::WALLETDB, "Flushing %s\n", strFile);
-                int64_t nStart = GetTimeMillis();
+    // Don't flush if any databases are in use
+    for (auto it = env->mapFileUseCount.begin() ; it != env->mapFileUseCount.end(); it++) {
```

Comment:
It makes no difference to the compiled code since we're not using the return value of the incrementer. I'll change to `++it` if I have to retouch this branch, but otherwise it's not worth invalidating ACKs.

Reply:
Yes. Even better. Will change if I have to retouch.